openhtf/core/test_executor.py

```diff
@@ -150,7 +150,13 @@ def _thread_proc(self):
       # Full plug initialization happens _after_ the start trigger, as close to
       # test execution as possible, for the best chance of test equipment being
       # in a known-good state at the start of test execution.
-      self.test_state.plug_manager.initialize_plugs()
+      try:
+        self.test_state.plug_manager.initialize_plugs()
+      except Exception:  # pylint: disable=broad-except
+        # Exit early if plug initialization failed.
+        self.test_state.finalize_on_failed_plug_initialization(
+            phase_executor.ExceptionInfo(*sys.exc_info()))
+        return
```

Comment:
Looks good, can we apply the same thing when running the test start?

Reply:
Done.